pandas/io/parquet.py

```diff
@@ -104,19 +104,26 @@ def write(
             from_pandas_kwargs["preserve_index"] = index
 
         table = self.api.Table.from_pandas(df, **from_pandas_kwargs)
+
+        fs = get_fs_for_path(path)
         # write_to_dataset does not support a file-like object when
         # a directory path is used, so just pass the path string.
         if partition_cols is not None:
             self.api.parquet.write_to_dataset(
                 table,
                 path,
                 compression=compression,
+                filesystem=fs,
```

Comment:
What if the user provides a `filesystem`?

Reply:
Thanks for catching this. I realised this after looking at the tests that failed.

Perhaps, `filesystem` should be one of the keyword arguments instead of getting it from `kwargs`?

I think we need to update the doc to mention that `filesystem` is needed if the system can't obtain the underlying credentials from e.g. `~/.aws/credentials`?